dlib/type_safe_union/type_safe_union_kernel.h

```diff
@@ -111,7 +111,10 @@ namespace dlib
         // --------------------------------------------
 
         // member data
-        stack_based_memory_block<max_size> mem;
+        typename std::aligned_union<max_size, T1,T2,T3,T4,T5, 
```

Comment:
Yeah this is a good change.  It's been 10 years since C++11 and dlib requires C++11 anyway now.  So any C++11 features are fine to use.

Remove the `max_size` member though.  That's not needed with `std::aligned_union` anymore. Like just set the argument here to 0 and let it do it's thing.

Reply:
Oh, i didn't know you could do that with ```std::aligned_union```